pytest_inmanta/plugin.py

```diff
@@ -951,38 +986,40 @@ def dryrun_resource(
         assert ctx.status == status
         return ctx.changes
 
-    def dryrun_all(
-        self, run_as_root: bool = False, assert_create_or_delete: bool = False
-    ) -> Dict[Resource, HandlerContext]:
+    def dryrun_all(self, run_as_root: bool = False) -> Result:
         """
         Runs a dryrun for every resource.
         :param run_as_root: run the mock agent as root
-        :param assert_create_or_delete: assert that every resource will either be created or deleted.
         """
-        results = {}
-        for resource in self.resources.values():
-            ctx = self.dryrun(resource, run_as_root=run_as_root)
-            if ctx.changes:
-                results[resource] = ctx
-            if assert_create_or_delete:
-                assert "purged" in ctx.changes
-        return results
+        return Result(
+            {
+                r: self.dryrun(r, run_as_root=run_as_root)
+                for r in self.resources.values()
+            }
+        )
 
     def dryrun_and_deploy_all(
         self, run_as_root: bool = False, assert_create_or_delete: bool = False
-    ) -> Dict[str, Dict[Resource, HandlerContext]]:
+    ) -> List[Result]:
```

Comment:
I would make this a tuple and clarify in the docstring what the ordering means.
```suggestion
    ) -> "tuple[Result, Result, Result]"
```

Reply:
Or even better, use a dataclass
```
@dataclass
class DryrunResult:
    first_dryrun: Result
    deploy: Result
    dryrun: Result
```

You could optionally make it iterable if that's important.